### PR TITLE
azfile: SAS

### PR DESCRIPTION
**Pull request by souravgupta-msft**
- [x] The purpose of this PR is explained in this or a referenced issue.
- [x] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [x] Tests are included and/or updated for code changes.
- [ ] Updates to [CHANGELOG.md][] are included.
- [x] MIT license headers are included in each file.

[Azure/autorest.go]: https://github.com/Azure/autorest.go
[CHANGELOG.md]: https://github.com/Azure/azure-sdk-for-go/blob/main/CHANGELOG.md

### CODE REVIEW

File: sdk/storage/azfile/sas/account.go [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+
+// AccountResourceTypes type simplifies creating the resource types string for an Azure Storage Account SAS.
+// Initialize an instance of this type and then call its String method to set AccountSASSignatureValues' ResourceTypes field.
+type AccountResourceTypes struct {
+	Service, Container, Object bool
+}
+
+// String produces the SAS resource types string for an Azure Storage account.
+// Call this method to set AccountSASSignatureValues' ResourceTypes field.
+func (rt *AccountResourceTypes) String() string {
```
Comment (gapra-msft) [inline sdk/storage/azfile/sas/account.go:147]: why do we have a parseAccountPermission but no parseAccountResourceTypes?

File: sdk/storage/azfile/sas/account_test.go [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+			List:   true,
+			Create: true,
+		}, input: "rwdlc"},
+		{expected: AccountPermissions{
+			Read:   true,
+			Write:  true,
+			Delete: true,
+			List:   true,
+			Create: true,
+		}, input: "rwldc"},
```
Comment (gapra-msft) [inline sdk/storage/azfile/sas/account_test.go:60]: randomize the order here to test that it works

File: sdk/storage/azfile/sas/service.go [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+	Version            string    `param:"sv"`  // If not specified, this defaults to Version
+	Protocol           Protocol  `param:"spr"` // See the Protocol* constants
+	StartTime          time.Time `param:"st"`  // Not specified if IsZero
+	ExpiryTime         time.Time `param:"se"`  // Not specified if IsZero
+	SnapshotTime       time.Time
+	Permissions        string  `param:"sp"` // Create by initializing a SharePermissions or FilePermissions and then call String()
+	IPRange            IPRange `param:"sip"`
+	Identifier         string  `param:"si"`
+	ShareName          string
+	FilePath           string // Ex: "directory/FileName" or "FileName". Use "" to create a Share SAS.
```
Comment (gapra-msft) [inline sdk/storage/azfile/sas/service.go:31]: Do directories in file share have SAS too? if so would path be more generic a name to cover that case too?

Comment (gapra-msft) [reply sdk/storage/azfile/sas/service.go:31]: also if they dont support it now, maybe they might in the future, so might be good to think about how to future proof this?

**Review by gapra-msft — COMMENTED** [review]
left a few comments

**Review by tasherif-msft — APPROVED** [review]

File: sdk/storage/azfile/sas/account.go (hunk 1 above)
Comment (souravgupta-msft) [reply sdk/storage/azfile/sas/account.go:153]: Added `parseAccountResourceTypes()` method.

File: sdk/storage/azfile/sas/account_test.go (hunk 2 above)
Comment (souravgupta-msft) [reply sdk/storage/azfile/sas/account_test.go]: Done

File: sdk/storage/azfile/sas/service.go (hunk 3 above)
Comment (souravgupta-msft) [reply sdk/storage/azfile/sas/service.go:31]: I think SAS is supported for directories as well. I checked the .NET SDK and they do have method to generate SAS in the directory client. What we can do is rename this parameter to `DirectoryOrFilePath`. Thoughts?

**Review by gapra-msft — APPROVED** [review]

**Pull request merged**